### DISCUSSION TITLE
How do you insert environment variable into cli_config.json file ?

### DISCUSSION THREAD

**Discussion by samixchoumi** [ryanrosello-og/playwright-slack-report#128] [Q&A]
Hello, I'm actually using this command : `npx playwright-slack-report --config="./cli_config.json" --json-results="./merged_tests_results.json"` to do the Slack reporting. But as I'm using the json file option, how can I use environment variable on it ?

Like, how can I call my CI env variable here : 
```
"meta": [
  { "key": "branch", "value" : "$CI_COMMIT_BRANCH"},
  { "key": "commit", "value" : "$CI_COMMIT_SHA"}
]
```
In the `.gitlab-ci.yml` it works properly, in the `playwright.config.ts` file too. But for the `.json` file, I don't find a solution as I though it would be compiled by node.

Is their a possibility to replace the `cli_config.json` to as `cli_config.ts` file ? If yes, is their an example to see how it works ?

**Comment by samixchoumi** [comment]
I found a solution using `jq` command before calling the `playwright-slack-report` command but it doesn't seem to be the easiest. If anyone have another solution, feel free to share.

**Reply by samixchoumi** [reply]
If anyone need a little example to adapt for their usage : 
`jq --arg yourArg "$YOUR_ENV_VAR" --arg anotherArg "$ANOTHER_ENV_VAR" '.sendUsingWebhook.webhookUrl |= $yourArg | .meta[] |= if .key == "Pipeline" then .value |= $anotherArg else . end' cli_config.json > cli_config.tmp.json`

Then you target cli_config.tmp.json file in your command : `npx playwright-slack-report --config="./cli_config.tmp.json" --json-results="./merged_tests_results.json"`

**Comment by ryanrosello-og** [comment]
yeah nothing easy at the moment. I feel like there should be an extra cli param e.g. 

`--meta="branch:$CI_COMMIT_BRANCH;commit:$CI_COMMIT_SHA"`

full cli would look something like this:


`npx playwright-slack-report --config="./cli_config.json" --json-results="./merged_tests_results.json" --meta="branch:$CI_COMMIT_BRANCH;commit:$CI_COMMIT_SHA"`

^ the meta cli param will take priority over the value in the values in config file

**Reply by samixchoumi** [reply]
Seem to be a nice improvement, if we still have the possibility to use both method it could be easier to do.
At least, I've sent an example of my solution to help.
